File: lab-1/src/lab1_pkg/scripts/relay.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from ackermann_msgs.msg import AckermannDriveStamped

class RelayNode(Node):
    def __init__(self):
        super().__init__('relay')
        self.subscription = self.create_subscription(
            AckermannDriveStamped,
            'drive',
            self.listener_callback,
            10)
        self.subscription  # prevent unused variable warning
        self.publisher_ = self.create_publisher(AckermannDriveStamped, 'drive_relay', 10)

    def listener_callback(self, msg):
        new_msg = AckermannDriveStamped()
        new_msg.drive.speed = msg.drive.speed * 3
        new_msg.drive.steering_angle = msg.drive.steering_angle * 3
        self.publisher_.publish(new_msg)
        self.get_logger().info('Publishing to /drive_relay: Speed: %f, Steering Angle: %f' % 
                               (new_msg.drive.speed, new_msg.drive.steering_angle))

def main(args=None):
    rclpy.init(args=args)
    relay = RelayNode()
    rclpy.spin(relay)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: lab-1/src/lab1_pkg/src/relay.cpp
#include "rclcpp/rclcpp.hpp"
#include "ackermann_msgs/msg/ackermann_drive_stamped.hpp"

class RelayNode : public rclcpp::Node {
public:
    RelayNode() : Node("relay") {
        subscription_ = this->create_subscription<ackermann_msgs::msg::AckermannDriveStamped>(
            "drive",
            10,
            std::bind(&RelayNode::listener_callback, this, std::placeholders::_1)
        );

        publisher_ = this->create_publisher<ackermann_msgs::msg::AckermannDriveStamped>("drive_relay", 10);
    }

private:
    void listener_callback(const ackermann_msgs::msg::AckermannDriveStamped::SharedPtr msg) {
        auto new_msg = ackermann_msgs::msg::AckermannDriveStamped();
        new_msg.header.stamp = this->get_clock()->now();
        new_msg.drive.speed = msg->drive.speed * 3;
        new_msg.drive.steering_angle = msg->drive.steering_angle * 3;

        publisher_->publish(new_msg);
        RCLCPP_INFO(this->get_logger(), "Publishing to /drive_relay: Speed: %f, Steering Angle: %f",
                    new_msg.drive.speed, new_msg.drive.steering_angle);
    }

    rclcpp::Subscription<ackermann_msgs::msg::AckermannDriveStamped>::SharedPtr subscription_;
    rclcpp::Publisher<ackermann_msgs::msg::AckermannDriveStamped>::SharedPtr publisher_;
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RelayNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}